src/source/Dtls/Dtls.c

```diff
@@ -44,11 +44,8 @@ STATUS dtlsTransmissionTimerCallback(UINT32 timerID, UINT64 currentTime, UINT64
     locked = TRUE;
 
     CHK(!SSL_is_init_finished(pDtlsSession->pSsl), STATUS_TIMER_QUEUE_STOP_SCHEDULING);
-
-    if (DTLSv1_get_timeout(pDtlsSession->pSsl, &timeout) == 0) {
-        // failed to get timeout. try again on next iteration
-        CHK(FALSE, retStatus);
-    }
+    // if not timeout yet. try again on next iteration
+    CHK(DTLSv1_get_timeout(pDtlsSession->pSsl, &timeout) == 1, retStatus);
```

Comment:
Not sure about checking part. The documentation doesn't say anything about the successful return being 1

https://boringssl.googlesource.com/boringssl/+/63fa118f3a3d065adb6ca17227bb9e407ffadccb/include/openssl/ssl.h

Perhaps, this should be CHK(DTLSv1_get_timeout(pDtlsSession->pSsl, &timeout) != 0, retStatus);